ui/src/services/utils.js

```diff
@@ -485,3 +485,47 @@ export const linuxDrivesNamingIncrement = (devicePath, increment) => {
     return '';
   }
 };
+
+// Return boolean to tell if the two alerts are the same only with different severity
+const isSameAlertWithDiffSeverity = (
+  alert1: $PropertyType<PrometheusAlert, labels>,
+  alert2: $PropertyType<PrometheusAlert, labels>,
+): boolean => {
+  // Filtering out the `severity` property
+  function replacer(key, value) {
+    if (key === 'severity') {
+      return undefined;
+    }
+    return value;
+  }
+
+  return JSON.stringify(alert1, replacer) === JSON.stringify(alert2, replacer);
+};
+
+/*
+ ** This function removes the warning alert from the list when critical one is triggered
+ ** It should be called at saga level before storing the alerts to `redux-store`
+ ** or where we resolve the promise with `react-query`
+ */
+export const removeWarningAlerts = (
+  alerts: Array<PrometheusAlert>,
+): Array<PrometheusAlert> => {
+  const len = alerts.length;
+  const removeIndex = [];
+  for (let i = 0; i < len - 1; i++) {
+    for (let j = i + 1; j < len; j++) {
+      if (isSameAlertWithDiffSeverity(alerts[i].labels, alerts[j].labels)) {
+        if (alerts[i].labels.severity === STATUS_WARNING) {
+          removeIndex.push(i);
+        } else if (alerts[j].labels.severity === STATUS_WARNING) {
+          removeIndex.push(j);
+        }
+      }
+    }
+  }
+
+  let removedWarningAlerts = alerts;
+  removeIndex.forEach((index) => removedWarningAlerts.splice(index, 1));
```

Comment:
In order to avoid any possible side effect on initial alerts array I would make `removedWarningAlerts` a copy of alerts instead of a simple ref. 
```suggestion
  let removedWarningAlerts = [...alerts];
  removeIndex.forEach((index) => removedWarningAlerts.splice(index, 1));
```